File: typical90/C++/026.cpp
#include <bits/stdc++.h>
using namespace std;
#define rep(i,n) for (int i = 0; i < (n); ++i)

int main() {
    int n;
    cin >> n;

    vector<vector<int>> G(n);
    rep(i,n-1) {
        int a, b;
        cin >> a >> b;
        a--; b--;
        G[a].push_back(b);
        G[b].push_back(a);
    }

    vector<int> color(n), one, two;
    auto dfs = [&](auto dfs, int pos, int cor) -> void {
        color[pos] = cor;
        if (cor%2) one.push_back(pos);
        else two.push_back(pos);
        for (int next_pos : G[pos]){
            if (color[next_pos]) continue;
            dfs(dfs, next_pos, 3-cor);
        }
    };
    dfs(dfs, 0, 1);
    
    if (one.size() > two.size()) {
        rep(i,n/2) cout << one[i]+1 << ' ';
    } else {
        rep(i,n/2) cout << two[i]+1 << ' ';
    }
    return 0;
}

File: typical90/Python/026.py
import sys
sys.setrecursionlimit(10 ** 9)

def dfs(pos, cor):
    color[pos] = cor
    for n_pos in G[pos]:
        if (color[n_pos] >= 1):
            continue
        dfs(n_pos, 3-cor)

n = int(input())

G = [[] for _ in range(n+1)]
for _ in range(n-1):
    a, b = map(int, input().split())
    G[a].append(b)
    G[b].append(a)

color = [0] * (n+1)
dfs(1,1)

one = list(filter(lambda i: color[i]%2==0, range(1,len(color))))
two = list(filter(lambda i: color[i]%2==1, range(1,len(color))))

print(*one[:n//2]) if len(one) > len(two) else print(*two[:n//2])